client/src/components/web-extension/Common/Account/useWallets.js

```diff
@@ -0,0 +1,71 @@
+import { useState, useEffect, useCallback } from 'react';
+import { WalletDescriptor } from 'casper-storage';
+import isNil from 'lodash-es/isNil';
+import get from 'lodash-es/get';
+
+import { getUserHDWallets, addWalletAccount } from '@cd/hooks/useServiceWorker';
+import { getAccounts } from '@cd/services/userServices';
+import { convertBalanceFromHex } from '@cd/helpers/balance';
+
+const formatAccountName = (index = 0) => {
+	return `Account ${index + 1}`;
+};
+
+export const useWallets = ({ isActive }) => {
+	const [wallets, setWallets] = useState([]);
+
+    const loadWalletsFromStorage = useCallback(async () => {
+        let hdWallets = await getUserHDWallets();
+		if (hdWallets.length === 0) {
+			await addWalletAccount(wallets.length, new WalletDescriptor(formatAccountName()));
+			hdWallets = await getUserHDWallets();
+		}
+
+		return hdWallets;
+    }, []);
+
+    const loadBalanceWallets = useCallback(async(hdWallets) => {
+        const publicKeys = hdWallets.map((wallet) => wallet.publicKey);
+		const accounts = await getAccounts(publicKeys);
+
+		const hdBalanceWallets = hdWallets.map((wallet) => {
+			const foundAccount = accounts.find((account) => account.publicKey === wallet.publicKey && isNil(wallet.balance));
+
+			return {
+				...wallet,
+				balance: foundAccount ? convertBalanceFromHex(foundAccount.balance.hex) : get(wallet, 'balance', 0),
+			};
+		});
+
+		return hdBalanceWallets;
+    }, [wallets]);
+
+	const loadWallets = useCallback(async () => {
+		const hdWallets = await loadWalletsFromStorage();
+		setWallets(oldWallets => {
+			return hdWallets.map(wallet => {
+				const foundOldWallet = oldWallets.find(oldWallet => oldWallet.publicKey === wallet.publicKey);
+
+				if (!foundOldWallet) {
+					return wallet;
+				}
+
+				return {
+					...foundOldWallet,
+					...wallet,
+				}
+			})
+		});
+
+        const hdBalanceWallets = await loadBalanceWallets(hdWallets);
+		setWallets(hdBalanceWallets);
+	}, [loadWalletsFromStorage, loadBalanceWallets]);
+
```

Comment:
Be careful with this approach, trigger loading data which depends on a boolean indicates a potential leak to infinite loop.
**For now we only show Wallet list inside a modal, that should be fine.**

When we want to show wallet list in a regular screen, there's a good chance we would reuse this hook. In this context a developer can accidentally pass `isActive` which is always `true`. This might trigger reloading and re-rendering parent or some other components. _In that case, he should use `loadWallets()` instead._